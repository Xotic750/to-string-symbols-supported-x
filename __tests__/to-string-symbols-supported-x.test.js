import noop from 'lodash/noop';
import toStr from 'src/to-string-symbols-supported-x';

/* eslint-disable-next-line compat/compat */
const hasSymbol = typeof Symbol === 'function' && typeof Symbol('') === 'symbol';
const ifSymbolIt = hasSymbol ? it : xit;

describe('basic tests', function() {
  it('should return a string for everything', function() {
    expect.assertions(1);
    const values = [true, 'abc', 1, null, undefined, noop, [], /r/];
    const expected = values.map(String);
    const actual = values.map(toStr);
    expect(actual).toStrictEqual(expected);
  });

  it('should throw for Object.create(null)', function() {
    expect.assertions(1);
    expect(function() {
      toStr(Object.create(null));
    }).toThrowErrorMatchingSnapshot();
  });

  ifSymbolIt('should return a string for Symbol', function() {
    expect.assertions(2);
    /* eslint-disable-next-line compat/compat */
    const sym = Symbol('foo');
    expect(toStr(sym)).toBe('Symbol(foo)');
    expect(toStr(Object(sym))).toBe('Symbol(foo)');
  });
});
