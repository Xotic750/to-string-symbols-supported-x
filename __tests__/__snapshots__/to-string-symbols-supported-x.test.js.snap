// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic tests should throw for Object.create(null) 1`] = `"Cannot convert object to primitive value"`;
